### PR TITLE
Add possible abi version to lock info

### PR DESCRIPTION
**Pull request by hoodmane**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+## [0.1.0a6] - 2024-04-03
+
+### Added
+
+- The `info` field now contains an optional `abi_version`.
```
Comment (ryanking13) [inline CHANGELOG.md:13]: The PR number is incorrect.

File: pyodide_lock/utils.py [hunk 2]
```diff
@@ -103,7 +103,7 @@ def _generate_package_hash(full_path: Path) -> str:
 
 
 def _get_marker_environment(
-    platform: str, version: str, arch: str, python: str
+    platform: str, version: str, arch: str, python: str, abi_version: str | None = None
```
Comment (ryanking13) [inline pyodide_lock/utils.py:106]: The parameter doesn't seem to be used anywhere?

**Review by ryanking13 — COMMENTED** [review]
Thanks!

File: pyodide_lock/utils.py (hunk 2 above)
Comment (hoodmane) [reply pyodide_lock/utils.py:106]: But it gets passed and will crash if we don't accept it.

Comment (ryanking13) [reply pyodide_lock/utils.py:106]: I see... we should probably update the contest.py file then.

**Pull request merged**